### ISSUE TITLE
Spyder opens issue reporter and /usr/local/bin/spyder on start

### ISSUE THREAD

**Issue by MathieuSch-101** [spyder-ide/spyder#14470]
### What steps will reproduce the problem?

I just did a clean install of python 3.8 with homebrew and of all my packages with pip install --force-reinstall pkg. The reason was that I could not get spyder to update previously, and even if pip list shown me spyder up to date with 4.2.1, spyder would still show a prompt stating that spyder was not up to date.

Everything works, even spyder, except that on every start, the file /usr/local/bin/spyder is opened (containing :)

    #!/usr/local/opt/python@3.8/bin/python3.8
    # -*- coding: utf-8 -*-
    import re
    import sys
    from spyder.app.start import main
    if __name__ == '__main__':
        sys.argv[0] = re.sub(r'(-script\.pyw|\.exe)?$', '', sys.argv[0])
        sys.exit(main())
       
And the popup 'Issue Reporter' opens too. The console shows: 'Layer-backing can not be explicitly controlled on 10.14 when build against the 1.0.14 SDK. 

The issue occurs both when starting spyder with:

    python /usr/local/lib/python3.8/site-packages/spyder/app/start.py
    
or with:

    /usr/local/bin/spyder
    
On the prevoious version, I was always using the second method with an alias in my zshrc configuration. Btw, the alias python refers to:

    /usr/local/opt/python@3.8/bin/python3

    

### Traceback
```python-traceback
  File "/usr/local/lib/python3.8/site-packages/spyder/plugins/editor/widgets/status.py", line 113, in process_git_data
    branches, branch, files_modified = output
TypeError: cannot unpack non-iterable NoneType object
```

## Versions

* Spyder version: 4.2.1 
* Python version: 3.8.6
* Qt version: 5.12.10
* PyQt5 version: 5.12.3
* Operating System: Darwin 19.6.0

### Dependencies

```

# Mandatory:
applaunchservices >=0.1.7       :  0.2.1 (OK)
atomicwrites >=1.2.0            :  1.4.0 (OK)
chardet >=2.0.0                 :  4.0.0 (OK)
cloudpickle >=0.5.0             :  1.6.0 (OK)
diff_match_patch >=20181111     :  20200713 (OK)
intervaltree >=3.0.2            :  3.1.0 (OK)
IPython >=7.6.0                 :  7.19.0 (OK)
jedi =0.17.2                    :  0.17.2 (OK)
jsonschema >=3.2.0              :  3.2.0 (OK)
keyring >=17.0.0                :  21.5.0 (OK)
nbconvert >=4.0                 :  6.0.7 (OK)
numpydoc >=0.6.0                :  1.1.0 (OK)
parso =0.7.0                    :  0.7.0 (OK)
pexpect >=4.4.0                 :  4.8.0 (OK)
pickleshare >=0.4               :  0.7.5 (OK)
psutil >=5.3                    :  5.8.0 (OK)
pygments >=2.0                  :  2.7.3 (OK)
pylint >=1.0                    :  2.6.0 (OK)
pyls >=0.36.2;<1.0.0            :  0.36.2 (OK)
pyls_black >=0.4.6              :  0.4.6 (OK)
pyls_spyder >=0.3.0             :  0.3.0 (OK)
qdarkstyle >=2.8                :  2.8.1 (OK)
qtawesome >=0.5.7               :  1.0.2 (OK)
qtconsole >=5.0.1               :  5.0.1 (OK)
qtpy >=1.5.0                    :  1.9.0 (OK)
setuptools >=39.0.0             :  51.1.0.post20201221 (OK)
sphinx >=0.6.6                  :  3.4.0 (OK)
spyder_kernels >=1.10.1;<1.11.0 :  1.10.1 (OK)
textdistance >=4.2.0            :  4.2.0 (OK)
three_merge >=0.1.1             :  0.1.1 (OK)
watchdog >=0.10.3               :  1.0.2 (OK)
zmq >=17                        :  20.0.0 (OK)

# Optional:
cython >=0.21                   :  None (NOK)
matplotlib >=2.0.0              :  3.3.3 (OK)
numpy >=1.7                     :  1.19.4 (OK)
pandas >=1.1.1                  :  1.1.5 (OK)
scipy >=0.17.0                  :  1.5.4 (OK)
sympy >=0.7.3                   :  None (NOK)
```

**Comment by MathieuSch-101** [comment]
Feel free to change the title to whatever it should be. I could not see the traceback before creating the git issue.

**Comment by ccordoba12** [comment]
Hey @MathieuSch-101, thanks for reporting. Unfortunately, we don't have the resources nor the time to help people with custom installations (in your case pip through Homebrew). So it's up to you to figure this one out.

Our supported methods are Anaconda and our installers, which you can find in our Releases page.

**Comment by ccordoba12** [comment]
Hey @MathieuSch-101, this seems an issue with Anaconda as well, according to issue #14483. So we'll fix it in our next version (4.2.2), to be released in mid February.
